### PR TITLE
Document kubevirt specific annotations and labels

### LINKED ISSUE #1080: Create a place to document annotations
With at least #1047 we have an annotation which is useful atm.

It would be good to have a place where we document annotations - just for developers. Thus not in the user-guide.

### PR DESCRIPTION
**Pull request by booxter**
**What this PR does / why we need it**: document annotations / labels specific to kubevirt.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1080 

**Special notes for your reviewer**: I was not there when all of these labels / annotations were added and hence may miss some crucial details. Feel free to point out any mistakes you can find. Some annotations / labels seem duplicative on first sight. I would be glad to learn if they are indeed duplicates and hence can be consolidated, or there is special reason for each of them to exist.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by slintes** [comment]
I find it useful, but the challenge of course is to keep the doc up to date...

**Comment by booxter** [comment]
/hold

Holding until we figure out in the issue what we actually want to see here. I understand and agree with the argument that internal annotations may have better place in the code; that being said, our parent project, kubernetes, documents it outside the code, so it seemed to make sense to follow their example.

File: docs/devel/annotations.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+## kubevirt.io/owned-by
+
+Example: kubevirt.io/owned-by=virt-handler
+
+Used on: Pod
+
+This annotation defines which KubeVirt component owns the resource. Similar to
+kubevirt.io.
+
+## kubevirt.io/nodeName
```
Comment (rmohr) [inline docs/devel/annotations.md]: This one is a hack because with CRDs we can't use field selectors. Internal.

File: docs/devel/annotations.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Example: kubevirt.io/nodeName=node02
+
+Used on: VirtualMachineInstance
+
+This label describes which cluster node runs the virtual machine instance.
+
+## kubevirt.io/schedulable
```
Comment (rmohr) [inline docs/devel/annotations.md]: This one is also important for the user.

File: docs/devel/annotations.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Example: kubevirt.io/schedulable=true
+
+Used on: Node
+
+This label declares whether a particular node is available for scheduling
+virtual machine instances on it.
+
+## kubevirt.io/heartbeat
```
Comment (rmohr) [inline docs/devel/annotations.md]: This one is also important for the user.

File: docs/devel/annotations.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+As of the time of writing, KubeVirt doesn't expose any user visible annotations
+or labels specific to its resources. That being said, KubeVirt components rely
+on a number of internal annotations and labels set on different resources to
+support its features. They will be listed below.
+
+**Note:** Annotations and labels that are not specific to KubeVirt are documented
+[here](https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/).
+
+## kubevirt.io
```
Comment (rmohr) [inline docs/devel/annotations.md]: This one is not used for ownership. We kind of have the rule to mark all kubevirt components (not vms), with the `kubevirt.io` key. The value (in this case `virt-launcher`), is a more or less optional description. The label can for instance be used to clean the cluster from kubevirt.

File: docs/devel/annotations.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Example: kubevirt.io=virt-launcher
+
+Used on: Pod
+
+This label defines which KubeVirt component owns the resource.
+
+## kubevirt.io/domain
```
Comment (rmohr) [inline docs/devel/annotations.md]: Internal.

File: docs/devel/annotations.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+Example: kubevirt.io/domain=vm-alpine-multipvc
+
+Used on: Pod
+
+This label can be used to match virtual machine instances represented as
+libvirt XML domains with their pods. Among other things, the label is used to
+detect virtual machines with dead pods.
+
+## kubevirt.io/created-by
```
Comment (rmohr) [inline docs/devel/annotations.md]: Internal.

File: docs/devel/annotations.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+Example: kubevirt.io/created-by=c323f009-7ef9-11e8-be52-525500d15501
+
+Used on: Pod
+
+This annotation is used to match virtual machine instance IDs with pods.
+Similar to kubevirt.io/domain.
+
+## kubevirt.io/owned-by
```
Comment (rmohr) [inline docs/devel/annotations.md]: Internal.

**Comment by SchSeba** [comment]
Hi please to a look in this [PR](https://github.com/kubevirt/kubevirt/issues/1087) we may need to add more labels to the document.

**Comment by booxter** [comment]
The latest upload moves all the documentation into inline comments in types.go. I also pushed a separate PR for user-guide to document the three labels / annotations that Roman suggested are of interest to users: https://github.com/kubevirt/user-guide/pull/102

**Comment by booxter** [comment]
@SchSeba note this PR no longer creates any documents, it just adds some comments. There is another PR for user-guide but that one documents just those labels / annotations useful to API consumers.

**Comment by booxter** [comment]
retest this please

**Comment by booxter** [comment]
@vladikr it's not a critical thing but it would be nice to have it off my plate. It shouldn't introduce any instability because it's just comments.

**Pull request merged**